drivers/overlay/overlay.go

```diff
@@ -1121,22 +1178,26 @@ func (d *Driver) getLower(parent string) (string, error) {
 }
 
 func (d *Driver) dir(id string) string {
-	p, _ := d.dir2(id)
+	p, _, _ := d.dir2(id)
 	return p
 }
 
-func (d *Driver) dir2(id string) (string, bool) {
+func (d *Driver) dir2(id string) (string, string, bool) {
 	newpath := path.Join(d.home, id)
+	imageStore := ""
+	if d.imageStore != "" {
+		imageStore = path.Join(d.imageStore, id)
```

Comment:
So we now have an “image store” variable, which is an individual per-image value, but `d.imageStore` is a per-store global.

And this function creates this `{s.imageStore}/{id}` value, but never checks it for any purpose. Why?
- If `{d.home}/{id}` exists, this function returns that as the first return value
- `{d.home}/{id}` does not exist but one of the “additional image stores” contain an `/{id}` path, this function returns that additional one.

What role does `{s.imageStore}/{id}` play here?

Before, this function fairly clearly returned “which directory contains the image data if the image exists, or where the data should be created if it doesn’t exist anywhere; is the image in an additional store”.

Now it returns … ???

In particular the various parent lookups only check in the `{s.imageStore}/{id}` location, outright failing if the image is not there. But this function does not check _at all_ whether anything exists in that location.